pkg/api/message/publishWorker.go

```diff
@@ -94,7 +94,12 @@ func (p *publishWorker) start() {
 		select {
 		case <-p.ctx.Done():
 			return
-		case new_batch := <-p.listener:
+		case new_batch, ok := <-p.listener:
+			if !ok {
+				p.log.Error("listener is closed")
```

Comment:
who restarts this after a failure? Is it a failure?

Reply:
```
go func() {
		s.poll()

		timer := time.NewTimer(s.options.Interval)
		for {
			timer.Reset(s.options.Interval)
			select {
			case <-s.ctx.Done():
				s.log.Debug("Context done; stopping subscription")
				close(s.updates)
				return
			case <-s.options.Notifier:
				s.poll()
			case <-timer.C:
				s.poll()
			}
		}
	}()
```

The listener should only get closed during shutdown. Which is the correct change.